source/user/mcscf.rst

```diff
@@ -1,57 +1,299 @@
-Although it is a triplet state, the solution might not be correct since the
-CASSCF is based on the incorrect singlet HF ground state.  Starting from the
-ROHF ground state, we have::
+Like many other modules in PySCF, :mod:`mcscf` works with density-fitting (:numref:`user_df`) and x2c (:numref:`user_x2c`).
+It can also be used in its unrestricted formulation, please see :source:`examples/mcscf/60-uhf_based_ucasscf.py` for an example.
+An important feature of :mod:`mcscf` is that it can interface with external CI solver such as DMRG, FCIQMC, or selected CI methods, see the external projects for more details.
 
-  >>> mc = mcscf.CASSCF(rhf3, 4, 6)
-  >>> print('E(CASSCF) = %.9g' % mc.kernel()[0])
-  E(CASSCF) = -149.646746
 
-The energy is lower than the RHF initial guess.
-.. We can also use the UHF ground
-.. state to start a CASSCF calculation::
-.. 
-..   >>> mc = mcscf.CASSCF(uhf3, 4, 6)
-..   >>> print('E(CASSCF) = %.9g' % mc.kernel()[0])
-..   E(CASSCF) = -149.661324
-..   >>> print('S^2 = %.7f, 2S+1 = %.7f' % mcscf.spin_square(mc))
-..   S^2 = 3.9713105, 2S+1 = 4.1091656
-.. 
-.. Woo, the total energy is even lower.  But the spin is contaminated.
+
+Picking an Active Space
+-----------------------
+In general, selecting an active space can be cumbersome and PySCF offers severals ways to facilitate this process.
+There are four strategies and all require that you specify the number of electrons and orbitals in the active space.
+
+.. warning::
+  We always advise users to visualize their chosen active orbitals before starting large/expensive calculations.
+  This involves dumping the MO coefficients to a ``molden`` file (see example :source:`examples/tools/02-molden.py`) and visualizing with your chosen program.
+  While there are many great softwares available to visualize orbitals, `JMol <http://jmol.sourceforge.net/>`_ is one of the easiest to use and is recommended for less experienced users.
+
+
+1) (Default) Specifying no additional information.
+  This is the most minimal strategy for selecting an active space and chooses orbitals (and electrons) around the fermi-level that match the number of orbitals and electrons specified by the user.
+  In most circumstances, this is not an ideal strategy and will lead to poor convergence or none at all.
+  For example:
+.. code-block:: python
+
+  ncas, nelecas = (6,8)
+  mycas = myhf.CASSCF(ncas, nelecas)
+
+
+2) Specifying the molecular orbital (MO) index of the active space orbitals you want. 
+The user can "manually" select the MO orbital indices (in a 1-based indexing scheme) and pass them to the ``sort_mo`` function.
+  See :source:`examples/mcscf/10-define_cas_space.py` for more details.
+
+.. code-block:: python
+
+  mycas = mcscf.CASSCF(myhf, 4, 4)
+  # Note sort_mo by default take the 1-based orbital indices.
+  mo = mycas.sort_mo([5,6,8,9])
+  mycas.kernel(mo)
+
+
+3) Specifying the number of orbitals in each symmetry group. 
```

Comment:
The symmetry composition of inactive, active and virtual spaces is fixed during the optimization. By default it is defied by starting orbitals.

Reply:
@bogdanoff, is this something you want added or are you saying you don't think strategy 3) is a useful one?

Reply:
I am going to edit this PR to resolve these comments, let's freeze further comments.

Reply:
I'm about to push my work from today so you may want to wait for 10 or so minutes